distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java

```diff
@@ -65,10 +65,9 @@ static List<String> systemJvmOptions(Settings nodeSettings, final Map<String, St
                 "-Dlog4j2.disable.jmx=true",
                 "-Dlog4j2.formatMsgNoLookups=true",
                 /*
-                 * Due to internationalization enhancements in JDK 9 Elasticsearch need to set the provider to COMPAT otherwise time/date
-                 * parsing will break in an incompatible way for some date patterns and locales.
+                 * Specify SPI to load IsoCalendarDataProvider (see #48209), specifying the first day of week as Monday
                  */
-                "-Djava.locale.providers=SPI,COMPAT",
+                "-Djava.locale.providers=SPI,CLDR",
```

Comment:
Should this change be conditionalized on the java version, ie only starting in 23?